PlayList.java

```diff
@@ -30,101 +32,201 @@ public Track getTrack(int index) {
             return null;
         }
     }
-    
-    /** Appends the given track to the end of this list. 
-     *  If the list is full, does nothing and returns false.
-     *  Otherwise, appends the track and returns true. */
+
+    /**
+     * Appends the given track to the end of this list.
+     * If the list is full, does nothing and returns false.
+     * Otherwise, appends the track and returns true.
+     */
     public boolean add(Track track) {
         //// replace the following statement with your code
+        if (this.size == this.maxSize) { // checks if have free space.
+            return false;
+        } else {
+            this.tracks[size] = track; // if there a space add track and make new space.
+            this.size = this.size + 1;
+        }
```

Comment:
`else` not required when the scope of `if` end with return
```suggestion
            this.tracks[size] = track; // if there a space add track and make new space.
            this.size = this.size + 1;
```